sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIO.java

```diff
@@ -781,11 +817,29 @@ public Write withFailureMode(FailureMode failureMode) {
       return toBuilder().setFailureMode(failureMode).build();
     }
 
-    /** Specifies the cell mutation limit. */
+    /** Specifies the cell mutation limit (maxumum number of mutated cells per batch). */
     public Write withMaxNumMutations(long maxNumMutations) {
       return toBuilder().setMaxNumMutations(maxNumMutations).build();
     }
 
+    /**
+     * Specifies an input PCollection that can be used with a {@code Wait.on(signal)} to indicate
+     * when the database schema is ready. To be used when the schema creation is part of the
+     * pipeline to prevent the connector reading the schema too early.
+     */
+    public Write withSchemaReadySignal(PCollection signal) {
```

Comment:
AFAIK, using [Wait.on(PCollection)](https://beam.apache.org/releases/javadoc/2.7.0/org/apache/beam/sdk/transforms/Wait.html) is the only way to 'pause' one pipeline until the results of a dependent section of the pipeline is complete. 
The given PCollection is the Signal to be passed to Wait.on()
Callbacks would not work because this needs to be triggered as part of the running pipeline. 

This transform reads the schema on start-up (line 914), if the schema is created by a separate section of the pipeline, there would be a race condition between creating the schema, and this transform reading it.

Improved javadoc with a link to Wait.OnSignal and made it clearer when/why this would be used.
This API is also described in the class level Javadoc `Database Schema Preparation`